Sources/BraintreeCore/Analytics/BTAnalyticsService.swift

```diff
@@ -103,52 +103,52 @@ class BTAnalyticsService: Equatable {
-
-            // A special value passed in by unit tests to prevent BTHTTP from actually posting
-            if let http = self.http, http.customBaseURL?.absoluteString == "test://do-not-send.url" {
-                return
-            }
+        // TODO: - Refactor to make HTTP non-optional property and instantiate in init()
+        if self.http == nil {
+            self.http = BTHTTP(authorization: self.apiClient.authorization, customBaseURL: BTAnalyticsService.url)
+        }
+        
+        // A special value passed in by unit tests to prevent BTHTTP from actually posting
+        if let http = self.http, http.customBaseURL?.absoluteString == "test://do-not-send.url" {
+            return
+        }
+        
+        if shouldBypassTimerQueue {
+            await self.sendQueuedAnalyticsEvents()
+            return
+        }
+        
+        if BTAnalyticsService.timer == nil {
+            BTAnalyticsService.timer = DispatchSource.makeTimerSource(queue: self.http?.dispatchQueue)
+            BTAnalyticsService.timer?.schedule(
+                deadline: .now() + .seconds(self.timerInterval),
+                repeating: .seconds(self.timerInterval),
+                leeway: .seconds(1)
+            )
             
-            if shouldBypassTimerQueue {
-                await self.sendQueuedAnalyticsEvents(configuration: configuration)
-                return
-            }
-
-            if BTAnalyticsService.timer == nil {
-                BTAnalyticsService.timer = DispatchSource.makeTimerSource(queue: self.http?.dispatchQueue)
-                BTAnalyticsService.timer?.schedule(
-                    deadline: .now() + .seconds(self.timerInterval),
-                    repeating: .seconds(self.timerInterval),
-                    leeway: .seconds(1)
-                )
-
-                BTAnalyticsService.timer?.setEventHandler {
-                    Task {
-                        await self.sendQueuedAnalyticsEvents(configuration: configuration)
-                    }
+            BTAnalyticsService.timer?.setEventHandler {
+                Task {
+                    await self.sendQueuedAnalyticsEvents()
                 }
-
-                BTAnalyticsService.timer?.resume()
             }
-        } catch {
-            return
+            
+            BTAnalyticsService.timer?.resume()
         }
     }
 
     // MARK: - Helpers
 
-    func sendQueuedAnalyticsEvents(configuration: BTConfiguration) async {
+    func sendQueuedAnalyticsEvents() async {
         if await !BTAnalyticsService.events.isEmpty {
-            let postParameters = await createAnalyticsEvent(config: configuration, sessionID: apiClient.metadata.sessionID, events: BTAnalyticsService.events.allValues)
-            http?.post("v1/tracking/batch/events", parameters: postParameters) { _, _, _ in }
-            await BTAnalyticsService.events.removeAll()
+            do {
+                let configuration = try await apiClient.fetchConfiguration()
```

Comment:
Would we also want to consider setting a property for this in the class and using an `if let` to continue to reuse the config if not nil? That could let us reuse the config for the entire flow vs fetching it every 20 seconds.

Reply:
The concern there is if the config had expired, we would be using an expired config

Reply:
Since it's stored as a singleton in the `ConfigurationCache` couldn't we check that? Psudocode but something like:
```swift
if let configuration, configuration = try? ConfigurationCache.shared.getFromCache(authorization: self.authorization.bearer)
```

Reply:
But honestly, does that even matter? Let's say a merchant uses our SDK with multiple merchantIDs (ie refreshing their clientToken would possible use a new merchantID). They would have to re-create a BTAPIClient with that refreshed clientToken which would create a new instance of BTAnalyticsService anyways.

Reply:
Yes, but in cases where merchants don't, which I believe is more common, it would avoid an additional hop to `fetchConfiguration` which while minor could reduce some latency. It is a take it or leave it comment so if you don't agree we don't need to implement it, was just asking a question.

Reply:
I was pairing with Rich and we observed a crash if we comment out first call to sendAnalyticsEvent to PayPalClient's tokenize function in branch where I created a merchant integration scenario where triple calls were made. I think we want to avoid making this call in sendAnalyticsEvent.

Reply:
Oh weird! My comment "But honestly, does that even matter?" was in response to questioning my previous comment [here](https://github.com/braintree/braintree_ios/pull/1337#discussion_r1643099274) 😆 . I must not have refreshed to see your comment, my apologies!

> Since it's stored as a singleton in the ConfigurationCache couldn't we check that? Psudocode but something like:

We _could_ but I don't think we want to leak logic of checking the config cache into the analytics layer. I think @richherrera is working on moving to the ConfigurationLoader pattern, like Android, which we can call in here once that's ready.

Reply:
Rich and I are pairing on working on the ConfigurationLoader pattern like you had suggested.

Reply:
> We could but I don't think we want to leak logic of checking the config cache into the analytics layer. I think @richherrera is working on moving to the ConfigurationLoader pattern, like Android, which we can call in here once that's ready.

That makes sense, we can punt on it until the ConfigurationLoader pattern is established!